queries/haskell/highlights.scm

```diff
@@ -11,6 +11,26 @@
 
 (comment) @comment
 
+((comment) @comment.documentation (#set! "priority" 101)
+  (#lua-match? @comment.documentation "^-- |"))
+
+((comment) @_first @comment.documentation (#set! "priority" 101)
+ (comment) @comment.documentation 
+  (#lua-match? @_first "^-- |"))
+
+((comment) @comment.documentation (#set! "priority" 101)
+  (#lua-match? @comment.documentation "^-- %^"))
+
+((comment) @_first @comment.documentation (#set! "priority" 101)
+ (comment) @comment.documentation 
+  (#lua-match? @_first "^-- %^"))
```

Comment:
This is not good.
For a chain of n comments node, the first will be matched n times, the second n-1 times, the third n-2 ... -> n^2 matches for n nodes.

Not to mention, this will also be matched by cases like this

```
(comment)
(n_random_arbitrary_nodes)
(comment)
```

Something like this is better
```
((comment) @_first . (comment)+ @comment.documentation)
```

Reply:
:thinking: if I anchor with `(comment)+` or `(comment)*`, it only captures the last `comment`.
I'll have to revisit this later or remove them.

Reply:
> 🤔 if I anchor with `(comment)+` or `(comment)*`, it only captures the last `comment`. I'll have to revisit this later or remove them.

It's an issue with neovim (neovim/neovim#24738). But yes, the comment queries you provided here will lead to performance issues, unless it's anchored and quantified.

Better to comment them out and re-look at a later date